lib/msf/core/exploit/remote/kerberos/service_authenticator/base.rb

```diff
@@ -0,0 +1,275 @@
+    client_name = username
+
+    tgt_result = send_request_tgt(
+      server_name: server_name,
+      client_name: client_name,
+      password: password,
+      realm: realm,
+    )
+
+    if !tgt_result.preauth_required
+      raise ::Rex::Proto::Kerberos::Model::Error::KerberosError.new(
+        'Kerberos ticket does not require preauthentication. It is not possible to decrypt the encrypted message to request further TGS tickets. Try cracking the password via AS-REP Roasting techniques.',
+      )
+    end
+
+    print_status("#{peer} - Valid TGT-Response")
+
+    now = Time.now.utc
+    expiry_time = now + 1.day
+
+    options = Rex::Proto::Kerberos::Model::KdcOptionFlags.from_flags(
+      [
+        Rex::Proto::Kerberos::Model::KdcOptionFlag::FORWARDABLE,
+        Rex::Proto::Kerberos::Model::KdcOptionFlag::RENEWABLE,
+        Rex::Proto::Kerberos::Model::KdcOptionFlag::CANONICALIZE,
+        Rex::Proto::Kerberos::Model::KdcOptionFlag::RENEWABLE_OK,
+      ]
+    )
+
+    # TODO: From [MS-KILE]:
+    #     The subkey in the EncAPRepPart of the KRB_AP_REP message (defined in [RFC4120] section 5.5.2) is used as the
+    #     session key when MutualAuthentication is requested. When DES and RC4 are used, the implementation is as defined
+    #     in [RFC1964]. With DES and RC4, the subkey in the KRB_AP_REQ message can be used as the session key, as it is
+    #     the same as the subkey in KRB_AP_REP message. However, when AES is used (see [RFC4121]), the subkeys are different
+    #     and the subkey in the KRB_AP_REP message is used. (The KRB_AP_REQ message is defined in [RFC4120] section 5.5.1).
+    #   So for now, we set the subkey to nil
+    subkey = nil
+
+    tgs_res = send_request_tgs(
+      req: build_tgs_request(
+        {
+          session_key: tgt_result.decrypted_part.key,
+          subkey: subkey,
+          checksum: nil,
+          ticket: tgt_result.ticket,
+          realm: realm,
+          client_name: client_name,
+          options: options,
+
+          body: build_tgs_request_body(
+            cname: nil,
+            sname: sname,
+            realm: realm,
+            options: options,
+
+            # Specify nil to ensure the KDC uses the current time for the desired starttime of the requested ticket
+            from: nil,
+            till: expiry_time,
+            rtime: nil,
+
+            # certificate time
+            ctime: now,
+          )
+        }
+      )
+    )
+
+    # Verify error codes
+    if tgs_res.msg_type == Rex::Proto::Kerberos::Model::KRB_ERROR
+      raise ::Rex::Proto::Kerberos::Model::Error::KerberosError.new(res: tgs_res)
+    end
+
+    print_good("#{peer} - Valid TGS-Response")
+    cache = extract_kerb_creds(
+      tgs_res,
+      tgt_result.decrypted_part.key.value,
+      msg_type: Rex::Proto::Kerberos::Crypto::KeyUsage::TGS_REP_ENCPART_SESSION_KEY
+    )
+    path = store_loot('windows.kerberos', 'application/octet-stream', rhost, cache.encode)
```

Comment:
Should this be `windows.kerberos`? It's not in the windows-specific kirbi format. Even though we're only really focusing on Windows support, should this instead be like `mit.kerberos.ccache`?

Reply:
I'll update this to be `mit.kerberos.ccache` :+1:

I grabbed the previous ltype from the kerberos checksum module:

```
$ rg 'store_loot.*kerberos' .
./modules/auxiliary/admin/kerberos/ms14_068_kerberos_checksum.rb
145:    path = store_loot('windows.kerberos', 'application/octet-stream', rhost, cache.encode)

./modules/post/multi/gather/unix_kerberos_tickets.rb
109:      loot_file = store_loot('unix_kerberos_tickets', 'application/octet-stream', session, data, "unix_kerberos_tickets_#{file}", 'Kerberos Tickets File')

./lib/msf/core/exploit/remote/kerberos/service_authenticator/base.rb
184:    path = store_loot('mit.kerberos.ccache', 'application/octet-stream', rhost, cache.encode)
```